astropy/tests/runner.py

```diff
@@ -124,12 +141,14 @@ def __new__(cls, *args, **kwargs):
 
     def _generate_args(self, **kwargs):
         # Update default values with passed kwargs
-        self.keywords.update(kwargs)
+        # but don't modify the defaults
+        keywords = copy.deepcopy(self.keywords)
+        keywords.update(kwargs)
         # Iterate through the keywords (in order of priority)
         args = []
-        for keyword in self.keywords.keys():
+        for keyword in keywords.keys():
             func = getattr(self, keyword)
-            result = func(self.keywords[keyword], self.keywords)
+            result = func(keywords[keyword], self.keywords)
```

Comment:
@Cadair - The `self.keywords` here should be `keywords` right ?
I don't have the time to test extensively now, but it seems to fix the issue (#5798).